### PR TITLE
Several refactorings in makefile.shade.

### PR DESCRIPTION
**Pull request by cesarbs**
- Read all environment variables at the top
- `camelCase` variables names instead of `LIKE_THIS`
- Reasonable defaults
- Only try to cache Coherence if the share exists

cc @victorhurdugaci

### CODE REVIEW

File: makefile.shade [hunk 1]
```diff
@@ -1031,6 +1034,28 @@ functions
                 }
             }
         }
+        
+        static string GetEnvironmentParameter(string name)
```
Comment (victorhurdugaci) [inline makefile.shade]: I think these should go in `KoreBuild`

File: makefile.shade [hunk 2]
```diff
@@ -17,29 +17,42 @@ use import="Json"
...
+        static string koreBuildTargets = GetEnvironmentParameter("KOREBUILD_BUILD_TARGETS");
+        static string coherenceFeed = GetEnvironmentParameter("COHERENCE_FEED");
+        static string koreBuildVersion = GetEnvironmentParameter("KOREBUILD_VERSION");
+        static string nugetVersion = GetEnvironmentParameter("NUGET_VERSION");
+        static string nugetExe = GetEnvironmentParameter("PUSH_NUGET_EXE");
+        static string preReleaseTag = GetEnvironmentParameter("PRERELEASETAG");
+        static string universeCommitsFile = GetEnvironmentParameter("UNIVERSE_COMMITS_FILE");
+        static bool skipNoCredentials = GetEnvironmentParameter("UNIVERSE_SKIP_NO_CREDENTIALS", value => value == "1");
 
         // Doesn't build on Mono since their contracts don't match
```
Comment (victorhurdugaci) [inline makefile.shade]: I think the comment is not for those two lines

File: makefile.shade [hunk 3]
```diff
@@ -17,29 +17,42 @@ use import="Json"
...
-        static string DROPS_SHARE = Environment.GetEnvironmentVariable("ASPNETCI_DROPS_SHARE") ?? @"\\aspnetci\drops";
-        static bool SKIP_NO_CREDENTIALS = Environment.GetEnvironmentVariable("UNIVERSE_SKIP_NO_CREDENTIALS") == "1";
+        static string baseDir = Directory.GetCurrentDirectory();
+        static string targetDir = Path.Combine(baseDir, "artifacts", "build");
+
+        static string buildBranch = GetEnvironmentParameter("BUILD_BRANCH", DefaultTo(DefaultBuildBranch));
+        static string coherenceCacheDir = GetEnvironmentParameter("COHERENCE_CACHE_DIR", DefaultTo(Path.Combine(baseDir, DefaultCoherenceCacheDir)));
+        static string dropsShare = GetEnvironmentParameter("ASPNETCI_DROPS_SHARE", DefaultDropsShare);
+        static string kBuildVersion = GetEnvironmentParameter("DNX_BUILD_VERSION");
+        static string ciVolatileShare = GetEnvironmentParameter("CI_VOLATILE_SHARE");
```
Comment (victorhurdugaci) [inline makefile.shade]: Wow, we're so consistent at naming environment variables

File: makefile.shade (hunk 1 above)
Comment (cesarbs) [inline makefile.shade]: https://github.com/aspnet/KoreBuild/pull/41

**Pull request merged**